language-plutus-core/src/Language/PlutusCore/Evaluation/Result.hs

```diff
@@ -22,6 +22,16 @@ data EvaluationResultF a
     | EvaluationFailure
     deriving (Show, Eq, Functor, Foldable, Traversable)
 
+instance Applicative EvaluationResultF where
```

Comment:
is there any reason we use `EvaluationResultF` over `Maybe`? Perhaps we could simply use a `newtype` wrapper here.

Reply:
For clarity and the `Pretty` instance mostly. Though, we can still use a `newtype` wrapper I guess. Thanks, I'll try that out.

Reply:
Hm, but then we probably need pattern synonyms and `-Wall` always complains about them being used in patterns.

Reply:
Is this fixable with `{-# COMPLETE -#}` pragmas?

Reply:
Dunno. I do not want this PR to be blocked on this. I'm not satisfied by the current approach, but the `Maybe` solution is not without drawbacks. It seems to me `EvaluationResult` does deserve its own constructors, but I need to think about it more. I'm resolving the issue, but also opening a ticket in my local task tracker.